app/models/operatingsystems/redhat.rb

```diff
@@ -25,9 +25,22 @@ def pxe_type
     "kickstart"
   end
 
+  def pxe_file_names(medium_provider)
+    if medium_provider.try(:architecture_name) =~ /^[Ss]390/
+      {
+        kernel: "kernel.img",
+        initrd: "initrd.img",
+      }
+    else
+      family.constantize::PXEFILES
+    end
+  end
+
   def pxedir(medium_provider = nil)
-    if medium_provider.try(:architecture).try(:name) =~ /^ppc64/
+    if medium_provider.try(:architecture_name) =~ /^ppc64/
```

Comment:
Should this become a `case medium_provider&.architecture_name` statement instead?